Sources/Purchasing/StoreKitAbstractions/StoreProduct.swift

```diff
@@ -367,3 +376,130 @@ private extension StoreProduct {
     }
 
 }
+
+/// Implementation of StoreProductType for web billing products
+private final class WebBillingStoreProduct: StoreProductType {
+    let product: WebProductsResponse.Product
+
+    init(product: WebProductsResponse.Product) {
+        self.product = product
+    }
+
+    var productCategory: StoreProduct.ProductCategory {
+        switch self.product.productType {
+        case "subscription":
+            return .subscription
+        default:
+            return .nonSubscription
+        }
+    }
+
+    var productType: StoreProduct.ProductType {
+        switch self.product.productType {
+        case "subscription":
+            return .autoRenewableSubscription
+        case "consumable":
+            return .consumable
+        case "non_consumable":
+            return .nonConsumable
+        default:
+            Logger.error("Unknown web product type: \(self.product.productType)")
+            return .autoRenewableSubscription
+        }
+    }
+
+    var localizedDescription: String {
+        return self.product.description ?? ""
+    }
+
+    var localizedTitle: String {
+        return self.product.title
+    }
+
+    var currencyCode: String? {
+        return self.defaultPurchaseOption?.basePrice?.currency ??
+            self.defaultPurchaseOption?.base?.price.currency
+    }
+
+    var price: Decimal {
+        if let basePrice = self.defaultPurchaseOption?.basePrice {
+            return Decimal(basePrice.amountMicros) / 1_000_000
+        } else if let base = self.defaultPurchaseOption?.base {
+            return Decimal(base.price.amountMicros) / 1_000_000
+        }
+        return 0
+    }
+
+    var localizedPriceString: String {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .currency
+        if let currencyCode = self.currencyCode {
+            formatter.currencyCode = currencyCode
+        }
+        return formatter.string(from: self.price as NSDecimalNumber) ?? "\(self.price)"
```

Comment:
I think you can use here the `priceFormatter` defined below
```suggestion
        self. priceFormatter.string(from: self.price as NSDecimalNumber) ?? "\(self.price)"
```